ui/src/logic/beaconchaApi/index.ts

```diff
@@ -0,0 +1,74 @@
+import { BeaconchaGetResponse } from "./types";
+import { ApiParams } from "../../types";
+
+//TODO Use inheritance for both APIs?
+export class BeaconchaApi {
+  baseUrl: string;
+  endpoint: string;
+  fullUrl: string;
+
+  constructor(apiParams: ApiParams) {
+    this.endpoint = apiParams.apiPath;
+    this.baseUrl = apiParams.baseUrl;
+    this.fullUrl = this.baseUrl + this.endpoint;
+  }
+
+  /*
+   * Fetch info for every validator PK
+   */
+  public async fetchAllValidatorsInfo({
+    beaconchaApi,
+    allValidatorPKs,
+  }: {
+    beaconchaApi: BeaconchaApi;
+    allValidatorPKs: string[];
+  }): Promise<BeaconchaGetResponse[]> {
+    const validatorsInfo = new Array<BeaconchaGetResponse>();
+
+    allValidatorPKs.forEach(async (pubkey) => {
```

Comment:
`forEach` is not meant for asyncronous programming, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach

Instead you should use other methods such as `for await (const x of y)`

Reply:
Done